localstack/services/sns/provider.py

```diff
@@ -1007,10 +1020,11 @@ def create_sns_message_body(subscriber, req_data, message_id=None):
         # Hardcoded
         "Signature": "EXAMPLEpH+..",
         "SigningCertURL": "https://sns.us-east-1.amazonaws.com/SimpleNotificationService-0000000000000000000000.pem",
+        "UnsubscribeURL": unsubscribe_url,
     }
 
     for key in ["Subject", "SubscribeURL", "Token"]:
-        if req_data.get(key):
+        if key in req_data and req_data[key][0]:
```

Comment:
The previous condition checked if it is available and truthy. Your changed code could raise a `TypeError` or an `IndexError` if `req_data` holds the key with value `None` or an empty list.

Reply:
The previous condition was always right, because it contains a list with None inside, which is not an empty list. It would add a `Subject: null` to the response.
But I see the issue now. This code was quite smart. I'll let the first check be `req_data.get(key)` but also check the first entry in the list is truthy.